app/main/routes.py

```diff
@@ -0,0 +1,209 @@
+    """
+    if not BucketList.query.all():
+        return errors.not_found("You currently have no BucketList saved.")
+
+    bucketlists = []
+
+    user = User.query.filter_by(username='andela').first().user_id
+
+    if request.args.get('q'):
+        try:
+            result = BucketList.query.filter_by(created_by=user).filter(
+                BucketList.name.contains(request.args.get('q'))).first()
+        except:
+            return errors.not_found(
+                "No BucketList exist with the name {}"
+                .format(request.args.get('q'))
+            )
+        finally:
+            bucketlists = result.to_json()
+
+    elif request.args.get('limit'):
+        pass
+    else:
+        for bucketlist in BucketList.query.filter_by(created_by=user):
+            bucketlists.append(bucketlist.to_json())
+
+    return jsonify({'bucketlist': bucketlists}), 200
+
+
+@main.route('/bucketlists/<int:list_id>', methods=['GET'])
+@json
+def get_bucketlist(list_id):
+    """
+    Get single bucket list.
+
+    Return a json of all the information as regards a particular BucketList.
+    """
+    bucketlist = BucketList.query.filter_by(bucketlist_id=list_id).first()
+    if not bucketlist:
+        return errors.not_found("The BucketList with the id: {0} doesn't"
+                                " exist.".format(list_id))
+    return bucketlist, 200
+
+
+@main.route('/bucketlists/', methods=['POST'])
+@json
+def create_bucketlist():
+    """
+    Create a new BucketList.
+
+    Add a new bucketlist and returns the bucketlist for the user to view
+    """
+    if not request.json or 'name' not in request.json:
+        return errors.bad_request("Only JSON object is accepted. Please "
+                                  "confirm that the key 'name' exists.")
+
+    user = User.query.filter_by(username='andela').first().user_id
+    try:
+        bucketlist = BucketList(name=request.json.get('name'), created_by=user)
+        bucketlist.save()
+    except:
+        return errors.bad_request("An error occurred while saving. "
+                                  "Please try again.")
+    return bucketlist, 201
+
+
+@main.route('/bucketlists/<int:list_id>', methods=['PUT'])
+@json
+def update_bucketlist(list_id):
+    """
+    Update BucketList.
+
+    Update a BucketList name.
+    """
+    bucketlist = BucketList.query.get_or_404(list_id)
+
+    if not request.json:
+        abort(400)
+
+    if 'done' not in request.json or 'name' not in request.json:
+        abort(400)
+
+    if type(request.json['name']) != unicode:
+        abort(400)
+
+    bucketlist.name = request.json.get('name')
+    bucketlist.save()
+    return bucketlist, 200
+
+
+@main.route('/bucketlists/<int:list_id>', methods=['DELETE'])
+def delete_bucketlist(list_id):
+    """
+    Delete a BucketList.
+
+    Deletes a BucketList and all items associated with it.
+    """
+    bucketlist = BucketList.query.get_or_404(list_id)
+
+    if not bucketlist:
+        abort(404)
```

Comment:
I think you shouldn't have this if statement here. `bucketlist` will never be a `None`, hence that `abort` will never be reached.  Check http://flask-sqlalchemy.pocoo.org/2.1/api/#flask.ext.sqlalchemy.BaseQuery.get_or_404